### ISSUE TITLE
Rewriting Weight Values

### ISSUE THREAD

**Issue by pervesh7707** [Haosam/Binary-Neural-Network-Keras#2]
Hello, I have successfully executed code, extracted weights, converted into binary, again assigned binary values to layer, but I am unable to update/write (binary weights) value to  h5 file.

**Comment by Haosam** [comment]
@pervesh7707 
Yes, I apologise as this is a single one time process, there is no fine tuning in the binarization
(Run model -> Binzarize -> re-Run model -> Binarize -> etc...), I am assuming you are asking this?
I used the premise after obtaining the 98% accuracy as the best-fine tuned model and binarized the results using a threshold set by my project professor.
I also apologise that I will not be continuing on this project as I am working full time now

**Comment by pervesh7707** [comment]
Thank you for your response, As per your statement and code, and my understanding,  run a model with full precision, and then Binarize it, repeat. As long as I understand the model is run with FP,  I want to check the model with binarized values.  
 pardon me, my question is "What is the scope of binarized weights/values then?"
I would love to read if you have any publications/documentation related to this? Regards

**Comment by Haosam** [comment]
@pervesh7707 
You can check this link out [binary neural network](https://arxiv.org/abs/1602.02830)
The work of this was initially to imitate the work initially done by Matthieu Courbariaux, Itay Hubara, Daniel Soudry, Ran El-Yaniv, Yoshua Bengio. The AI greats.
As their implementation of the code was done on Theano and Lasagne (A now depreciated library), my original task was to implement what they did onto Java. But given the complexity of what they had done, a lower level language (Java) and I did not have experience in Java; I had to compromise to redo this code onto Keras/Tensorflow.
As this was an academic project and not production code, I pretty much did an edit of this person, [dingke](https://github.com/DingKe/BinaryNet)

*You may read their work and search what they have done

The initial focus of this work was not so much of the code, but my professor was developing a on-chip training device to increase the 'rate' of writing onto the chip while attempting to decrease the energy usage by >100x (I think)
*You may look at this, Year 2019, Section 3 [Prof Kim Bongjin](https://sites.google.com/view/mslab/publications?authuser=0)
- He did not publish the paper on the site though

I believe the code was further enhanced by the PhD student as he was working on Theano and Pytorch as he did tell me he was able to pull out each individual hidden layer. But I do apologize as I have not followed his work, neither do I know his github since I graduated.

**Comment by Haosam** [comment]
In a very simplistic view, they wanted to maintain accuracy of object detection while reducing power consumption of the ALUs (Specialized chips dedicated for neural networks), so that these chips may 1 day be deployed onto small devices like cameras, mobile phones or even smaller than that.
The last I heard they were working on the CIFAR10 dataset, but that's all I know as I do not keep up to date with their work and I have other priorities too.

Apart from that, I will be closing this issue now. I apologize once more that I have not been prompt on my replies as I had no plans on maintaining this project.
